lib/loadGltfUris.js

```diff
@@ -1,134 +1,99 @@
 'use strict';
+var Cesium = require('cesium');
+var Jimp = require('jimp');
+var Promise = require('bluebird');
+var dataUriToBuffer = require('data-uri-to-buffer');
 var fs = require('fs');
 var path = require('path');
-var Cesium = require('cesium');
+
 var defined = Cesium.defined;
 var defaultValue = Cesium.defaultValue;
 var DeveloperError = Cesium.DeveloperError;
+
 var isDataUri = require('./isDataUri');
-var dataUriToBuffer = require('data-uri-to-buffer');
-var async = require('async');
-var Jimp = require('jimp');
 
 module.exports = loadGltfUris;
 
-function loadGltfUris(gltf, options, callback) {
-    var basePath = options.basePath;
-
-    var wrappedCallback = function(err, gltf) {
-        gltf.extras._pipeline.jimpScratch = new Jimp(1, 1, function(jimpErr, image) {
-            if (jimpErr) {
-                throw jimpErr;
-            }
-            gltf.extras._pipeline.jimpScratch = image;
-            callback(err, gltf);
+function loadGltfUris(gltf, options) {
+    return new Promise(function(resolve, reject) {
+        var basePath = options.basePath;
+        var loadURIs = [
+            loadURI(gltf, basePath, 'buffers'),
+            loadURI(gltf, basePath, 'images'),
+            loadURI(gltf, basePath, 'shaders')
+        ];
+        Promise.all(loadURIs).then(function() {
+            gltf.extras._pipeline.jimpScratch = new Jimp(1, 1, function(err) {
```

Comment:
Hm, I haven't looked into this at all but is it possible to promisify a constructor's callback?